keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -552,25 +553,52 @@ class ApplicationService(
     val releaseStatus = repository.getReleaseStatus(artifact, version)
     return repository.getArtifactVersion(artifact, version, releaseStatus)
   }
-}
 
-/**
- * Query the repository for all of the verification states associated with [versions]
- *
- * This just calls [KeelRepository.getVerificationStatesBatch] and reshapes the returned value to a map
- */
-fun KeelRepository.getVerificationStates(
-  deliveryConfig: DeliveryConfig,
-  versions: List<PublishedArtifact>
-): Map<VerificationContext, Map<Verification, VerificationState>> =
-  deliveryConfig.contexts(versions).let { contexts ->
-    contexts.zip(getVerificationStatesBatch(contexts))
-      .associate { (ctx: VerificationContext, vIdToState: Map<String, VerificationState>) ->
-        ctx to (vIdToState.entries.associate { (vId, state) ->
-          ctx.verification(vId) to state
-        })
+  /**
+   * Query the repository for all of the verification states associated with [versions]
+   *
+   * This just calls [KeelRepository.getVerificationStatesBatch] and reshapes the returned value to a map
+   */
+  fun KeelRepository.getVerificationStates(
+    deliveryConfig: DeliveryConfig,
+    versions: List<PublishedArtifact>
+  ): Map<VerificationContext, Map<Verification, VerificationState>> =
+    deliveryConfig.contexts(versions).let { contexts: List<VerificationContext> ->
+      contexts.zip(getVerificationStatesBatch(contexts)) // List<Pair<VerificationContext, Map<String, VerificationState>>
+        .associate { (ctx, vIdToState) -> ctx to vIdToState.toVerificationMap(deliveryConfig, ctx) }
+    }
+
+  /**
+   * Convert a (verification id -> verification state) map to a (verification -> verification state) map
+   *
+   * Most of the logic in this method is to deal with the case where the verification id is invalid
+   */
+  fun Map<String, VerificationState>.toVerificationMap(deliveryConfig: DeliveryConfig, ctx: VerificationContext) : Map<Verification, VerificationState> =
+    entries
+      .mapNotNull { (vId: String, state: VerificationState) ->
```

Comment:
I think a slightly more idiomatic way of doing this would be using `mapKeys` here and a `filterNotNull` at the end (avoids the need to convert back to a map at the end).

Reply:
It looks like there is no  built-in `filterNotNull` function defined on maps, only `filterNotNullValues`, where here what I want to do is filter out null keys.